app/controllers/tasks_controller.rb

```diff
@@ -1,53 +1,68 @@
 class TasksController < ApplicationController
-  load_and_authorize_resource :project
-  load_and_authorize_resource :task, through: :project
-
+  load_and_authorize_resource :only => [:my_tasks, :update_task_logtime]
+  load_and_authorize_resource :project, except: [:my_tasks, :update_task_logtime]
+  load_and_authorize_resource :task, through: :project, except: [:my_tasks, :update_task_logtime]
+ 
   def create
     @task.set_status
     if @task.save
-      redirect_to project_tasks_url(@task.project), notice: "Returing from the create"
+      redirect_to project_tasks_url(@task.project), notice: 'Returing from the create'
     else
-      render new_project_task_path(@task.project), notice: "Cannnot enter data due to constraints"
+      render new_project_task_path(@task.project), notice: 'Cannnot enter data due to constraints'
     end
   end
 
   def update
     if !params[:assignable_employee_id].empty? && !params[:assignable_team_id].empty?
-      render :edit, notice: "Make sure you delete employee field or team field"
+      render :edit, notice: 'Make sure you delete employee field or team field'
     else
-      params[:task][:status] = "new" if params[:task][:assignable_id].empty?
-      params[:task][:status] = "assigned" if !params[:task][:assignable_id].empty?
+      params[:task][:status] = 'new' if params[:task][:assignable_id].empty?
+      params[:task][:status] = 'assigned' if !params[:task][:assignable_id].empty?
       if @task.update(task_params)
-        redirect_to project_tasks_path, notice: "Updated Successfully"
+        redirect_to project_tasks_path, notice: 'Updated Successfully'
       else
-        redirect_to project_tasks_path, notice: "Cannot be updated"
+        redirect_to project_tasks_path, notice: 'Cannot be updated'
       end
     end
   end
 
   def destroy
     @task.destroy
     if @task.destroyed?
-      redirect_to project_tasks_path, notice: "Deleted Successfully"
+      redirect_to project_tasks_path, notice: 'Deleted Successfully'
     else
-      redirect_to project_tasks_path, notice: "Cannot be deleted successfully"
+      redirect_to project_tasks_path, notice: 'Cannot be deleted successfully'
     end
   end
 
   def update_status
     if @task.update(task_params)
-      redirect_to project_tasks_path, notice: "Updated Successfully"
+      redirect_to project_tasks_path, notice: 'Updated Successfully'
     else
-      redirect_to project_tasks_path, notice: "Cannot be updated!"
+      redirect_to project_tasks_path, notice: 'Cannot be updated!'
 
     end
   end
 
+  def update_task_logtime
+    unless params[:task][:log_time].blank?
+      if @task.log_time.blank?
+        @result = @task.update_attributes(log_time: params[:task][:log_time].to_i)
```

Comment:
you have done this using a lengthier approach.
`@task.update_attributes(log_time: @task.log_time + params[:task][:log_time].to_i)
`
^ this should work

Reply:
I guess this can be done using only update. Path would be different. Discuss!